semantic_model_generator/tests/cte_utils_test.py

```diff
@@ -320,3 +372,82 @@ def test_col_expr_object_type(self) -> None:
             match=f"We do not support object datatypes in the semantic model. Col {col.name} has data type {col.data_type}. Please remove this column from your semantic model or flatten it to non-object type.",
         ):
             _validate_col(col)
+
+    def test_enrich_column_in_expr_with_aggregation(self) -> None:
+        col_format_tbl = get_test_table_col_format_w_agg_only()
+        got = _enrich_column_in_expr_with_aggregation(col_format_tbl)
+        want = semantic_model_pb2.Table(
+            name="t1",
+            base_table=semantic_model_pb2.FullyQualifiedTable(
+                database="db", schema="sc", table="t1"
+            ),
+            columns=[
+                semantic_model_pb2.Column(
+                    name="d2_total",
+                    kind=semantic_model_pb2.ColumnKind.measure,
+                    description="d2_description",
+                    expr="sum(d2)",
+                ),
+                # Expand the column referred in expr.
+                semantic_model_pb2.Column(
+                    name="d2",
+                    expr="d2",
+                ),
+            ],
+        )
+        assert got == want
+
+    def test_enrich_column_in_expr_with_aggregation_and_renaming(self) -> None:
+        tbl = get_test_table_col_format_agg_and_renaming()
+        got = [c for c in _enrich_column_in_expr_with_aggregation(tbl).columns]
+        want = [
+            semantic_model_pb2.Column(
+                name="cost",
+                kind=semantic_model_pb2.ColumnKind.measure,
+                expr="cst",
+            ),
+            semantic_model_pb2.Column(
+                name="clicks",
+                kind=semantic_model_pb2.ColumnKind.measure,
+                expr="clcks",
+            ),
+            semantic_model_pb2.Column(
+                name="cpc",
+                kind=semantic_model_pb2.ColumnKind.measure,
+                expr="sum(cst) / sum(clcks)",
+            ),
+            semantic_model_pb2.Column(
+                name="clcks",
+                expr="clcks",
+            ),
+            semantic_model_pb2.Column(
+                name="cst",
+                expr="cst",
+            ),
+        ]
+        got.sort(key=lambda c: c.name.lower())
+        want.sort(key=lambda c: c.name.lower())
+        assert got == want
+
+    def test_expand_all_logical_tables_as_ctes(self) -> None:
+        vq = "SELECT * FROM __t2"
+        ctx = get_test_ctx_col_format()
```

Comment:
do we want to add a test including `sum(cst)/sum(clcks)` and making sure `cst`, `clck` are correctly reflected in CTE?

Reply:
Done.